src/TLuaInterpreter.cpp

```diff
@@ -3076,33 +3077,37 @@ int TLuaInterpreter::setBackgroundColor(lua_State* L)
     return 0;
 }
 
-int TLuaInterpreter::calcFontWidth(int size)
+int TLuaInterpreter::calcFontSize(lua_State* L)
 {
-    QFont font = QFont("Bitstream Vera Sans Mono", size, QFont::Normal);
-    return QFontMetrics(font).width(QChar('W'));
-}
+    Host* pHost = &getHostFromLua(L);
 
-int TLuaInterpreter::calcFontHeight(int size)
-{
-    QFont font = QFont("Bitstream Vera Sans Mono", size, QFont::Normal);
-    int fontDescent = QFontMetrics(font).descent();
-    int fontAscent = QFontMetrics(font).ascent();
-    return fontAscent + fontDescent;
-}
+    QString windowName = QStringLiteral("main");
+    QPair<int, int> size;
+
+    // pre- setFont(), miniconsoles were fixed to the Bitsteam font and so calcFontSize was fixed to it as well
+    if (lua_gettop(L) == 1 && !lua_isnumber(L, 1)) {
+        auto fontSize = lua_tonumber(L, 1);
+        auto font = QFont(QStringLiteral("Bitstream Vera Sans Mono"), fontSize, QFont::Normal);
+
+        auto fontMetrics = QFontMetrics(font);
+        auto width = fontMetrics.width(QChar('W'));
+        auto height = fontMetrics.ascent() + fontMetrics.descent();
```

Comment:
Um, according to the Qt Docs this is the same as `(int) QFontMetrics::height()` so why not use that instead and save on a *tiny* (admittedly) bit of maths...